### ISSUE TITLE
High WindowServer usage on Sequoia Beta 2

### ISSUE THREAD

**Issue by underscoregeb** [pablopunk/SwiftShift#68]
### Description
I'm seeing excessive CPU levels (up to 90%) from WindowServer while Swift Shift is running. This comes with poor system performance and low frame rates. It usually occurs anywhere from 10 to 30 minutes after Swift Shift is started. The issue persists in macOS Safe Mode.

I'd be happy to provide logs, but i'm not sure the best way to get them.

### Environment
MacBook Pro with M1, macOS Sequoia Beta 2 (24A5279h)
This may or may not be related to the beta, as I installed Swift Shift for the first time **after** updating to sequoia. (wonderful app, by the way!)

### Steps to reproduce
- Start Swift Shift
- Wait up to 30m

**Comment by pablopunk** [comment]
I guess this is the same issue as #47, which is prior to Sequoia. I run the app 24/7 and don't have any issues so let's see if the others that can reproduce it come up with a cause

**Comment by AndrewShaw123** [comment]
@pablopunk yes it's enabled. it seem normal after disable it. thank you

**Comment by pablopunk** [comment]
Thanks for confirming @AndrewShaw123. I'll try to debug what's causing the issue when it's enabled, since it makes the app unusable when that feature is on

**Comment by danstewart** [comment]
I've also been having this issue on Sequoia 15.3.1.

I can't figure out what triggers it, it doesn't seem very consistent but is always after running for a while.

Today on my personal computer it started lagging within 2 hours but on my work computer it usually doesn't start acting up until the afternoon after working for a good few hours.

A restart seems to sort it though.

Here are my settings:
![Image](https://github.com/user-attachments/assets/d698cbbf-73c2-4707-855f-55faf746dcd6)

**Comment by pablopunk** [comment]
hey guys @underscoregeb @AndrewShaw123 @danstewart if you still use the app can you update and see if the issue is gone?

**Comment by danstewart** [comment]
I'll give it a try tomorrow, thanks!

I've currently been working around the issue by having a script auto restart SwiftShift every two hours.

**Comment by danstewart** [comment]
@pablopunk I've had no issues today, thanks for fixing 🎉

**Comment by acruise** [comment]
FWIW I'm on 1.0.3 and have this issue, with "require mouse click" enabled because that's what I love from KDE (alt-left-drag to move, alt-right-drag to resize). On first launch everything seems fine; after an hour or two I get catastrophic input lag

**Comment by pablopunk** [comment]
Released a new version: 1.2.0

**Comment by acruise** [comment]
FYI it got quite a bit better in 1.2.2 but things still get laggy after some time (ten minutes? a couple hours?)

**Comment by marshalldevs** [comment]
I find this still happening I have to kill swiftshift and reopen it again every 30 minute when system is under load.

**Comment by pablopunk** [comment]
@marshalldevs do yo use the mouse button settings?

**Comment by marshalldevs** [comment]
Not quite sure what you mean by that. If you mean the move and resize yes I use left and right mouse button for them.

**Comment by pablopunk** [comment]
@marshalldevs yes that's what I meant. That's likely where the issue comes from. I'm investigating this also [here](https://github.com/pablopunk/SwiftShift/issues/167)

**Comment by pablopunk** [comment]
hopefully fixed in https://github.com/pablopunk/SwiftShift/pull/169 - will release now

_Update: 1.3.1. is out!_
